pkg/server/controllers.go

```diff
@@ -621,6 +621,52 @@ func (s *Server) installAPIBindingController(ctx context.Context, config *rest.C
 		return err
 	}
 
+	controllerName = "permission-claim-label-controller"
+	if err := server.AddPostStartHook(postStartHookName(controllerName), func(hookContext genericapiserver.PostStartHookContext) error {
+		logger := klog.FromContext(ctx).WithValues("postStartHook", postStartHookName(controllerName))
+		// do custom wait logic here because APIExports+APIBindings are special as system CRDs,
+		// and the controllers must run as soon as these two informers are up in order to bootstrap
+		// the rest of the system. Everything else in the kcp clientset is APIBinding based.
+		if err := wait.PollImmediateInfiniteWithContext(goContext(hookContext), time.Millisecond*100, func(ctx context.Context) (bool, error) {
+			crdsSynced := s.ApiExtensionsSharedInformerFactory.Apiextensions().V1().CustomResourceDefinitions().Informer().HasSynced()
+			exportsSynced := s.KcpSharedInformerFactory.Apis().V1alpha1().APIExports().Informer().HasSynced()
+			bindingsSynced := s.KcpSharedInformerFactory.Apis().V1alpha1().APIBindings().Informer().HasSynced()
+			return crdsSynced && exportsSynced && bindingsSynced, nil
+		}); err != nil {
+			logger.Error(err, "failed to finish post-start-hook")
+			return nil // don't klog.Fatal. This only happens when context is cancelled.
+		}
+
+		go permissionClaimLabelController.Start(goContext(hookContext), 5)
+
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	controllerName = "permission-claim-label-resource-controller"
```

Comment:
Do these controllers export their `.Name` as a package constant? If not, could they?

Reply:
Yeah, they do. I'll switch things over and make sure they're all pointing at the package vars and add those if they're not present.